doc/user/content/transform-data/temporal-filters.md

```diff
@@ -0,0 +1,368 @@
+
+```sql
+-- Maintained collection of only valid results.
+CREATE MATERIALIZED VIEW valid_events
+AS SELECT content, count(*)
+FROM events
+WHERE mz_now() >= insert_ts
+  AND mz_now() < delete_ts
+GROUP BY content;
+```
+
+What's all this then?
+
+Presumably `valid_events` has the property that if you `SELECT` from it you should see the same results as for the time-windowed `SELECT` in the previous section. That is 100% true.
+
+What is also true is that `valid_events` has enough information from you, in the form of the query itself, to maintain only enough historical detail to answer these `SELECT` queries from now onward. Once Materialize's `mz_now()` passes a record's `delete_ts` it cannot be seen (at least not through this view), and Materialize can dispose of the event. The in-memory footprint of `valid_events` stays bounded by the number of records in the system that could still satisfy this constraint (those records that are currently valid, or who may yet become valid in the future).
+
+While you add to `events`, Materialize collects up the events that are no longer visible, automatically. Of course, you can also ***change*** the records in `events`, in case you want to remove some events early, or draw out the `delete_ts` of any record, or replace one event with its next stage (and new `insert_ts` and `delete_ts`). If you happen to adjust any fields that interact with `mz_now()` Materialize will update the views appropriately.
+
+## A Brief Example
+
+Let's do some testing with our table and maintained view. Tables have the nice property that we can interactively update them from within Materialize, rather than spinning up a Kafka cluster.
+
+Let's start with something simple: we'll just look at the records currently present in our `valid_events` view. Let's define a different view to do that, though, without the aggregation so you can see the raw data:
+
+```sql
+-- Maintained collection of only valid results.
+CREATE MATERIALIZED VIEW valid AS
+SELECT content, insert_ts, delete_ts
+FROM events
+WHERE mz_now() >= insert_ts
+  AND mz_now() < delete_ts;
+```
+
+We'll print out the things in our view, along with the current logical timestamp. It is initially empty, because we haven't put any data in. But, these are the columns we'll be looking at.
+
+```
+materialize=> SELECT *, mz_now() FROM valid;
+
+content | insert_ts | delete_ts | mz_now
+---------+-----------+-----------+----------------------
+(0 rows)
+
+```
+
+Now let's put some data in there. I'm going to just take advantage of the fact that `INSERT` statements can also use `mz_now()` to populate the data with some records that we will make last five seconds.
```

Comment:
This statement became false, unfortunately. You cannot insert into table with `mz_now()` any more (you could when this was written).
```
Error: calls to mz_now in write statements are not supported
```

Reply:
We don't use it in the examples, and imo it sucks a bit that we use `now()` rather than `mz_now()`, because we spent some effort arguing that it was what MZ uses for time. But .. probably best to route around the damage and just not mention it.